daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -189,8 +203,9 @@ class Compiler private[japi] () {
    * @param namespace namespace of the root node. Set to empty string to specify
    *                  no namespace. Set to to NULL to figure out the namespace.
    */
+  @deprecated("Pass argument to compileSource, or compileFile.", "Since 2.6.0")
```

Comment:
arguments plural

Reply:
Probably shouldnt' deprecate this for now. Consistent with not changing JAPI/SAPI at all.

Reply:
Ok. I will take a pass at deprecating all setters everywhere on the basic API, the JAPI/SAPI and TDML processor APIs. 

The headache is updating all the tests that call these things now using what will be deprecated methods.

Reply:
If updating the test is too much of pain we could change the sbt config so that warnings are not treated as errors for the test target. So we would have have a different ``scalacOptions in Test :=`` in the sbt build settings.